$OpenBSD: patch-lib_TWiki_Client_TemplateLogin_pm,v 1.1 2011/05/19 06:27:30 jasper Exp $

Security fix for CVE-2011-1838:
XSS Vulnerability with origurl parameter of login script

Patch from upstream:
http://twiki.org/cgi-bin/view/Codev/SecurityAlert-CVE-2011-1838#Hotfix_for_TWiki_Production_Rele

--- lib/TWiki/Client/TemplateLogin.pm.orig	Thu May 19 08:24:32 2011
+++ lib/TWiki/Client/TemplateLogin.pm	Thu May 19 08:25:03 2011
@@ -107,6 +107,9 @@ sub login {
     my $loginName = $query->param( 'username' );
     my $loginPass = $query->param( 'password' );
 
+    # Item6673: Cleanup origurl parameter
+    $origurl   =~ s/[^a-zA-Z0-9_\-\.\:\/\?\;\&]//g;
+
     # Eat these so there's no risk of accidental passthrough
     $query->delete('origurl', 'username', 'password');
 
